src/inference/src/ie_core.cpp

```diff
@@ -988,7 +988,24 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
                         "set_property is supported only for BATCH itself (without devices). "
                         "You can configure the devices with set_property before creating the BATCH on top.");
 
-        ExtractAndSetDeviceConfig(properties);
+        bool isMetaDevice = device_name.find("AUTO") != std::string::npos ||
+                            device_name.find("MULTI") != std::string::npos ||
+                            device_name.find("HETERO") != std::string::npos;
+        if (!isMetaDevice) {
+            // unsupport to set ov::device::properties to HW device through this function
+            auto devices = GetListOfDevicesInRegistry();
+            for (auto&& config : properties) {
+                auto parsed = parseDeviceNameIntoConfig(config.first);
+                auto is_secondary_config_for_hw_device =
+                    std::any_of(devices.begin(), devices.end(), [&](const std::string& device) {
+                        return device == parsed._deviceName;
+                    });
+                OPENVINO_ASSERT(
+                    !is_secondary_config_for_hw_device,
+                    "set_property only supported ov::device::propreties for Meta device (AUTO/MULTI/HETERO). "
```

Comment:
Some properties are part of ov:: instead of ov::device::

Please change to “set_property only supports properties for specific Meta device (AUTO/MULTI/HETERO)”